### ISSUE TITLE
Process title corner case update bug (argv[0] scan of /proc/${pid}/cmdline) in htop 2.2.0

### ISSUE THREAD

**Issue by etosan** [hishamhm/htop#814]
Even with

 [x] Update process names on every refresh

On Linux `htop` 2.2.0 seems to be missing certain cases.

Most importantly it seems, it doesn't detect `agv[0]` (eg. proctitle) truncations.

When proctitle gets longer `htop` updates process title accordingly, but when `argv[0]` truncates,
htop still displays (by now) "old" title.

Linux `top` and `ps` from procps-ng 3.3.15 don't seem to have the issue.

This is important when hunting for stuck workers in worker swarms that truncate their names when processing multiple requests.

**Comment by Explorer09** [comment]
Could you please test the patch in pull request #812 ? Your report looks like a duplicate of that.

**Comment by etosan** [comment]
Explorer, will look into it tomorrow but indeed I think that is the error. This was my "fix":

```diff
@@ -669,6 +669,7 @@ static void LinuxProcessList_readDelayAcctData(LinuxProcessList* this, LinuxProc
 static void setCommand(Process* process, const char* command, int len) {
    if (process->comm && process->commLen >= len) {
       strncpy(process->comm, command, len + 1);
+      process->comm[len + 1] = 0;
    } else {
       free(process->comm);
       process->comm = xStrdup(command);
```

**Comment by hishamhm** [comment]
Indeed, it was fixed in #812 — thanks for reporting!
